ee/billing/billing_manager.py

```diff
@@ -167,13 +167,13 @@ def get_billing(self, organization: Optional[Organization], plan_keys: Optional[
             product["percentage_usage"] = current_usage / usage_limit if usage_limit else 0
 
             # Also update the tiers
-            if product["tiers"]:
+            if product.get("tiers"):
                 product["tiers"] = compute_usage_per_tier(current_usage, product["projected_usage"], product["tiers"])
                 product["current_amount_usd"] = sum_total_across_tiers(product["tiers"])
 
             # Update the add on tiers
             # TODO: enhanced_persons: make sure this updates properly for addons with different usage keys
-            for addon in product["addons"]:
+            for addon in product.get("addons"):
                 if not addon["subscribed"]:
```

Comment:
`product.get("addons")` will return `None` if `addons` is not an attribute of `product`. 

So if `addons` does not exist, the next line will be trying to iterate over None. Additionally, if `product["addons"]` is an empty list, line 177 will try call `[]['addons']` which would error as well. 

If product['addons'] returns a dictionary without the key `subscribed` it will throw a KeyError. 

Typically, what I do is create a guard clause that handles the expected data not existing and then continue down the happy path.

Example: 

```
if "addons" not in product:
   ... # handle it not existing

for addon in product["addons"]:
    if "subscribed" not in addon:
        ... # code that handles self hosting plebs like me xD
    if not addon["subscribed"]:
        ... # handle the logic you originally intended

Reply:
Absolutely! I'd go even further, where if you aren't sure of the shape of data, one case this still misses is when the key exists but is set to None / a falsy value.

I'd replace the above with 

```
for addon in (product.get("addons") or [])
```

Note that `product.get("addons", [])` wouldn't work, because the key exists, and will return `None` if the value is None.